src/Sdk/Common/Common/RawHttpMessageHandler.cs

```diff
@@ -0,0 +1,349 @@
+using GitHub.Services.OAuth;
+
+namespace GitHub.Services.Common
+{
+    public class RawHttpMessageHandler: HttpMessageHandler
+    {
+        public RawHttpMessageHandler(
+            VssOAuthCredential credentials)
+            : this(credentials, new RawClientHttpRequestSettings())
+        {
+        }
+
+        public RawHttpMessageHandler(
+            VssOAuthCredential credentials,
+            RawClientHttpRequestSettings settings)
+            : this(credentials, settings, new HttpClientHandler())
+        {
+        }
+
+        public RawHttpMessageHandler(
+            VssOAuthCredential credentials,
+            RawClientHttpRequestSettings settings,
+            HttpMessageHandler innerHandler)
+        {
+            this.Credentials = credentials;
+            this.Settings = settings;
+            m_messageInvoker = new HttpMessageInvoker(innerHandler);
+            m_credentialWrapper = new CredentialWrapper();
+
+            // If we were given a pipeline make sure we find the inner-most handler to apply our settings as this
+            // will be the actual outgoing transport.
+            {
+                HttpMessageHandler transportHandler = innerHandler;
+                DelegatingHandler delegatingHandler = transportHandler as DelegatingHandler;
+                while (delegatingHandler != null)
+                {
+                    transportHandler = delegatingHandler.InnerHandler;
+                    delegatingHandler = transportHandler as DelegatingHandler;
+                }
+
+                m_transportHandler = transportHandler;
+            }
+
+            ApplySettings(m_transportHandler, m_credentialWrapper, this.Settings);
+
+            m_thisLock = new Object();
+        }
+
+        /// <summary>
+        /// Gets the credentials associated with this handler.
+        /// </summary>
+        public VssOAuthCredential Credentials
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Gets the settings associated with this handler.
+        /// </summary>
+        public RawClientHttpRequestSettings Settings
+        {
+            get;
+            private set;
+        }
+
+        // setting this to WebRequest.DefaultWebProxy in NETSTANDARD is causing a System.PlatformNotSupportedException
+        //.in System.Net.SystemWebProxy.IsBypassed.  Comment in IsBypassed method indicates ".NET Core and .NET Native
+        // code will handle this exception and call into WinInet/WinHttp as appropriate to use the system proxy."
+        // This needs to be investigated further.
+        private static IWebProxy s_defaultWebProxy = null;
+
+        /// <summary>
+        /// Allows you to set a proxy to be used by all RawHttpMessageHandler requests without affecting the global WebRequest.DefaultWebProxy.  If not set it returns the WebRequest.DefaultWebProxy.
+        /// </summary>
+        public static IWebProxy DefaultWebProxy
```

Comment:
if you look at where we call `InitializeVssClientSettings` you will see we initialize the WebProxy based on the HTTP_PROXY and HTTPS_PROXY variables. This allows users to set custom proxies with settings, I think your pr is missing that code but I may have overlooked it.

Reply:
absolutely, I am pretty sure I left more things out, I wanted to do bare minimum to get this working and add more as we progress.

Reply:
https://github.com/github/c2c-actions-service/issues/3297